compiler/src/symboltable.rs

```diff
@@ -66,6 +66,7 @@ pub enum SymbolTableType {
     Module,
     Class,
     Function,
+    Comprehension,
```

Comment:
I would rather stick to the three options here:

https://docs.python.org/3/library/symtable.html?highlight=symboltable#symtable.SymbolTable.get_type

Still, it is weird that a comprehension is a function (which it is clearly not), but that's the way cpython did it ..

Reply:
There are some differences between the interface spec and the implementation. To achive compatiblity you need more context as the interface has. https://github.com/python/cpython/blob/7b78e7f9fd77bb3280ee39fb74b86772a7d46a70/Python/symtable.c#L1435 
There also some more flags that are not visible in the interface.